lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -3798,12 +3798,14 @@ prop_balanceTransactionValid wallet (ShowBuildable partialTx) seed
                 (SelectionBalanceErrorOf EmptyUTxO))) ->
                 label "empty UTxO" $ property True
             Left (ErrBalanceTxInternalError
-                 (ErrUnderestimatedFee delta candidateTx)) ->
+                 (ErrUnderestimatedFee delta candidateTx nWits)) ->
                 let counterexampleText = mconcat
                         [ "underestimated fee by "
                         , pretty delta
                         , "\n candidate tx: "
                         , pretty candidateTx
+                        , "\nassuming key witness count: "
+                        , show nWits
                         ]
```

Comment:
Suggestion: use `unlines` here to save yourself from having to manually insert "\n" and " " at the start and end of each line.

```suggestion
                let counterexampleText = unlines
                        [ "underestimated fee by:"
                        , pretty delta
                        , "candidate tx:"
                        , pretty candidateTx
                        , "assuming key witness count:"
                        , show nWits
                        ]
```